docs/discussion/Cosign Upgrade 2024.md

```diff
@@ -0,0 +1,600 @@
+  - multiple certificates definine in a single AKV KMS
+  - mix of keys + certificates in a single AKV KMS
+  - a way to specify specific keys/certificates within a KMS by name and version
+  - a way to scope a policy to a particular registry/repo
+  - multiple policies depending on registry scope
+  - a way to specify enforcement for that specific policy
+    - skip: don't perform any verification for an image reference that matches this policy
+    - any: at least one of the keys/certificates trusted in the policy must result in a successful verification for overall cosign verification to be true
+    - all: ALL keys/certificates trusted in the policy must result in a successful verification for overall cosign verification to be true
+  - a way to define certificates to be used in a trust policy for Trusted Timestamp verification `tsaCerts`
+  - a way to define options per trust policy for transparency log verification `tLogVerify`
+  - a way to define options per trust policy for keyless verification
+    - certificate transparency log lookup `ctLogVerify`
+  - a way to scope the policy based on `artifactType` if we have nested verification `artifactTypeScopes`
+    - if the user defines `artifactTypeScopes`, then the `registryScopes` are applied first and then `artifactTypeScopes`. `artifactTypeScopes` are only enforced if the subject image manifest being verified contains a `subject` field (basically is a referrer).
+
+Sample
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: Verifier
+metadata:
+  name: verifier-cosign
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-weight: "5"
+spec:
+  name: cosign
+  artifactTypes: application/vnd.dev.cosign.artifact.sig.v1+json
+  parameters:
+    trustPolicies:
+      - name: wabbit-networks-images
+        registryScopes:
+          - wabbitnetworks.myregistry.io/carrots
+          - wabbitnetworks.myregistry.io/beets
+        artifactTypeScopes: # OPTIONAL: applied after registryScopes; only considered for nested verification scenarios. 
+          - application/sarif+json
+          - application/spdx+json
+        keys: # list of keys that are trusted. Only the keys in KMS are considered
+          - provider: inline/inline-keys-1 # REQUIRED: if name is not provided, all keys are assumed to be trusted in KeyManagementProvider resource specified
+          - provider: inline/inline-keys-2
+          - provider: azurekeyvault/akv-wabbit-networks
+            name: wabbit-networks-io # OPTIONAL: key name
+            version: 1234567890 # OPTIONAL: key version (inline will not support version)
+        certificates: # list of certificates that are trusted. Only the certificates in KMS are considered
+          - provider: inline/inline-certs-1
+        tsaCerts:
+          - provider: inline/inline-certs-tsa-1
+        tLogVerify: true # transparency log verification (default to false)
```

Comment:
Should we put parameters for keyless under a parent parameter? For example,
```
trustPolicies:
 - name: 1
   - Registry scopes
   - artifact types
   - keys
   - keyless
   - enforcement
 - name: 2
    - ... 
```

Reply:
yes good idea

Reply:
I've added a section grouping all keyless configs